crates/sui-framework/sources/governance/validator_set.move

```diff
@@ -30,40 +31,40 @@ module sui::validator_set {
         quorum_stake_threshold: u64,
 
         /// The current list of active validators.
-        active_validators: vector<Validator>,
+        active_validators: VecMap<address,Validator>,
 
         /// List of new validator candidates added during the current epoch.
         /// They will be processed at the end of the epoch.
-        pending_validators: vector<Validator>,
+        pending_validators: VecMap<address, Validator>,
 
-        /// Removal requests from the validators. Each element is an index
-        /// pointing to `active_validators`.
-        pending_removals: vector<u64>,
+        /// Removal requests from the validators. Each element is the address of a validator in
+        /// `active_validators`. There are no duplicates
+        pending_removals: vector<address>,
 
         /// The metadata of the validator set for the next epoch. This is kept up-to-dated.
```

Comment:
This comment seems inconsistent with the current state of the code to me--`next_epoch_validators` only gets updated in `advance_epoch`. Which is the intended behavior?

If the intended behavior is "only update in `advance_epoch`", I think we probably do not need this at all (but could be wrong).

Reply:
ah good catch. This is intended to be updated each time a request is made. Must have forgot to add them.

Reply:
Cool, I will add a TODO/task